teuthology/packaging.py

```diff
@@ -369,3 +381,229 @@ def get_package_version(remote, package):
+                    )
+                    log.exception(msg)
+                    raise RuntimeError()
+            return codename
+
+        return "{distro}{version}".format(
+            distro=distro,
+            version=version,
+        )
+
+    def _get_uri(self):
+        """
+        Set the uri -- common code used by both install and debian upgrade
+        """
+        tag = branch = sha1 = None
+        if self.remote:
+            tag = _get_config_value_for_remote(self.ctx, self.job_config,
+                                               self.remote, 'tag')
+            branch = _get_config_value_for_remote(self.ctx, self.job_config,
+                                                  self.remote, 'branch')
+            sha1 = _get_config_value_for_remote(self.ctx, self.job_config,
+                                                self.remote, 'sha1')
+        else:
+            sha1 = self.sha1
+
+        if tag:
+            uri = 'ref/' + tag
+        elif branch:
+            uri = 'ref/' + branch
+        elif sha1:
+            uri = 'sha1/' + sha1
+        else:
+            # FIXME: Should master be the default?
+            log.debug("defaulting to master branch")
+            uri = 'ref/master'
+        return uri
+
+    def _get_base_url(self):
+        """
+        Figures out which package repo base URL to use.
+        """
+        template = config.baseurl_template
+        # get distro name and arch
+        base_url = template.format(
+            host=config.gitbuilder_host,
+            proj=self.project,
+            pkg_type=self.pkg_type,
+            arch=self.arch,
+            dist=self.distro,
+            flavor=self.flavor,
+            uri=self.uri,
+        )
+        return base_url
+
+    def _get_package_version(self):
+        """
+        Look for, and parse, a file called 'version' in base_url.
+        """
+        url = "{0}/version".format(self.base_url)
+        log.info("Looking for package version: {0}".format(url))
+        resp = requests.get(url)
+        version = None
+        if not resp.ok:
+            log.info(
+                'Package not there yet (got HTTP code %s), waiting...',
+                resp.status_code,
+            )
+        else:
+            version = resp.text.strip()
+            # TODO: move this parsing into a different function for
+            # easier testing
+            # FIXME: 'version' as retreived from the repo is actually the
+            # RPM version PLUS *part* of the release. Example:
+            # Right now, ceph master is given the following version in the
+            # repo file: v0.67-rc3.164.gd5aa3a9 - whereas in reality the RPM
+            # version is 0.61.7 and the release is 37.g1243c97.el6 (centos6).
+            # Point being, I have to mangle a little here.
+            if version[0] == 'v':
+                version = version[1:]
+            if '-' in version:
+                version = version.split('-')[0]
```

Comment:
I don't think you need the if condition here, because the split will give you one item if `'-'` is not there so it is safe to always split for it.

Not that this needs to change though, checking for it is OK too.